### ISSUE TITLE
Release version 2.0.0

### ISSUE THREAD

**Issue by cdubz** [babybuddy/babybuddy#515]
I have started work recently on a [v2 branch](https://github.com/babybuddy/babybuddy/tree/v2) and will try to focus future efforts on moving that forward with some issues with breaking changes. I'll use this issue on tracking for the relevant issues that need to be completed.

## Planned release date

26 May

### Breaking changes

- `TIME_ZONE` variable no longer supported. User timezone can be set from user settings.
- `NAP_START_MIN` and `NAP_START_MAX` variables no longer supported. Nap time parameters can be set from site settings.
- `Timer` model `end` and `duration` fields removed. There is no longer a concept of "active" vs. "inactive" timers.
- API: `active` filter on `Timer` model no longer supported.
- API: `PATCH /api/timers/$x/stop` endpoint no longer supported.

### Major changes

- Custom datetime selector has been removed. Browser defaults for locale datetime fields are now used.
- Timers no longer have an "end" and cannot be stopped -- only deleted or used to create a new entry with times.
- There is a new site-wide settings option under user menu.
- Sleep entries can now be defined as a "nap" (or not) manually.
- 24-hour time override has been removed. Time format will be determined by browser datetime fields based on locale.

## v2.0.0

- [x] #635
- [x] #301
- [x] #636
- [x] #637
- [x] #638
- [x] #109 
- [x] #639
- [x] #513 

## Stretch goals

- [x] #470 
- [ ] #471
- [ ] #460
- [ ] #161 
- [ ] #34

**Comment by whnr** [comment]
Will it be totally breaking or is the intention to be able to migrate data from v1 to v2?

**Comment by cdubz** [comment]
@whnr my intention is for there to be a migration path using regular database migrations. I'm not being _too_ ambitious here with the 2.0.0 tag so I'm fairly confident that will be achievable.

**Comment by cdubz** [comment]
@jcgoette @MrApplejuice @OttPeterR curious to get y'alls eyes on the v2 changes (see draft release notes above) w.r.t. your integrations!

**Comment by MrApplejuice** [comment]
Thanks! Yeah... the Android app will be "kaputt" when this comes live:

#109

It was anyway my intention to give this a rework this in the app. I would like to have some time to deploy a hotfix for this if at all possible. Is there a schedule for the major-release by any chance? I think that I can have a fix ready until next week!

**Comment by cdubz** [comment]
@MrApplejuice no real rush here. I’m happy to wait til those of you with known integrations are ready to roll.

**Comment by MrApplejuice** [comment]
Hey thanks! I still you should probably think about setting a deadline for the v2 release, otherwise you might wait a long time should projects become dormant. But nonetheless, thank you for the heads-up! I will try and fix my timer-controls asap!

**Comment by cdubz** [comment]
Yeah that’s fair. We’ve got a long vacation coming up so I’m hesitant to drop it and then disappear hah. My rough plan is to release near the end of the month. Let’s say 26 May 👍

**Comment by lenaxia** [comment]
I'll try to get it spun up this weekend. Looking forward to it!

**Comment by OttPeterR** [comment]
Appreciate the heads up! I think those breaking changes should be fine. I like moving more things to the settings rather than env vars. The timer changes seem interesting, would there be a way to use that new architecture to note that a toddler woke up in the middle of the night and not have to get rid of the original sleep timer that spanned the whole night?


As for timeline expectations, I don't have a lot of personal dev time in the next few weeks (still working on the 1.16.0 update). Unfortunately I might not be able to put sufficient time into this until early June.

**Comment by jcgoette** [comment]
Thanks for the heads up!

Are you still accepting PRs for this release? @Cadair has actually started working on fixing the HA integration, but I'm thinking using the new site-wide settings model to create an API endpoint exposing the site's BB version would be useful for my integration.

**Comment by cdubz** [comment]
@jcgoette sure thing!

**Comment by cdubz** [comment]
Oh oops also pinging @btoconnor re: https://github.com/babybuddy/babybuddy/issues/515#issuecomment-1536798129

**Comment by btoconnor** [comment]
@cdubz I should be able to look at this next week - thanks for the ping!

**Comment by MrApplejuice** [comment]
Alright! My app has been updated, I released it on GitHub and on the play-store it is currently "in review":

![image](https://github.com/babybuddy/babybuddy/assets/542105/df5067ca-4594-41d8-a5bf-a5d2e86389ac)

And here the GitHub release: https://github.com/MrApplejuice/BabyBuddyAndroid/releases/tag/v2.0.0

This was a crunch to get this out in time... had to change thousands of lines of code. Kind of my fault for not implementing the right level of abstraction from the get-go though! :shrug: 

Anyhow, it typically takes a few days for the app store to approve the change. So I would not mind if the release can wait a few more days... but might not be actually necessary. I guess it will take some time anyways until the update proliferates to other servers. Again... dunno :shrug: 

Thanks again for the tap on the back so that this could be fixed and rolled out in a timely manner!

**Comment by MrApplejuice** [comment]
Well, sorry for the spam, but: 

![image](https://github.com/babybuddy/babybuddy/assets/542105/eed19dae-47a4-4b6a-9ef2-67b0e143153c)

So, no need to wait for the Play Store from my side!

**Comment by cdubz** [comment]
Well 26 May came and went :smile: I'm still getting readjusted from a vacation and have a few issues to review that may slow down the 2.0.0 release. Takin' a look now :eyes:

**Comment by cdubz** [comment]
Ok it looks like the only thing I still want to poke around on is https://github.com/babybuddy/babybuddy/issues/644#issuecomment-1545644400. Will try to do that in the next few days and then move the release along.

**Comment by cdubz** [comment]
:wave: just pushed a v2.0.0-beta2 release taking care of some of the bugs that popped up. #644 is good again. Think this is pretty much there...

**Comment by btoconnor** [comment]
Just finding some time now to get a look at this - are there docker images for the beta releases? I'm not seeing them on https://hub.docker.com/r/linuxserver/babybuddy

I can set up something without docker, just a little easier for me to use the images there.

I still haven't found a great workflow for testing the Alexa integration for a parallel instance of the web server...

**Comment by cdubz** [comment]
@btoconnor can you use `latest` image to test? If not we could ask the LSIO folks if we can get beta release builds somehow.

**Comment by cdubz** [comment]
FTR I'm about ready for this -- just waiting on finalization in #674.

**Comment by cdubz** [comment]
And we're off! 🎉 

https://github.com/babybuddy/babybuddy/releases/tag/v2.0.0

**Comment by btoconnor** [comment]
Leaving some notes here even though the issue is closed. I finally got around to testing this out (sorry for delay).  Most of the actions seem to still work fine, except for concluding timers.  I'll hopefully get to fix that soon.  I'll leave further comments / issues on the alexa repo
